ondine/cards2-revised.js

```diff
@@ -0,0 +1,152 @@
+var cardTools = {
+
+	rank: function(card) {  // 1 ... 13
+    return this.isValid(card,0,51) &&
+      Math.floor(card/4) + 1;
+	},
+
+	suit: function(card) {  // 1 ... 4
+		return this.isValid(card,0,51) && 
+      (card%4) + 1;
+	},
+
+	cardID: function(rank,suit) {  // 0 ... 51
+		return this.isValid(rank, 1, 13) &&
+           this.isValid(suit, 1, 4) && 
+           (4 * (rank - 1) + (suit - 1));
+	},
+
+	color: function(card) {
+    var suitColor = this.suit(card);
+    return suitColor && ((suitColor < 3) ? "red": "black");
+	},
+
+	name: function(card) {
+    var cardRank = this.rank(card);
+    var cardSuit = this.suit(card);
+    return cardRank && cardSuit && (this.rankName[cardRank] + " of " + this.suitName[cardSuit]);
+	},
+
+  isValid: function(num, low, high) {
+    if ((typeof num) != "number") {
+      return NaN;
+    }
+    if ((num%1) !== 0) {
+      return NaN;
+    }
+    if (num < low || num > high) {
+      return NaN;
+    }
+    return true;
+  },
+
+  rankName: ["", "Ace", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten", "Jack", "Queen", "King"],
+  suitName: ["", "Hearts", "Diamonds", "Spades", "Clubs"]
+ 
+};
+
```

Comment:
The above looks good but you need to watch your formatting. Anything after '{ ' or a conditional should be indented. 
```
+	rank: function(card) {  // 1 ... 13
+    return this.isValid(card,0,51) &&
+      Math.floor(card/4) + 1;
+	},

```

should look like

```
+	rank: function(card) {  // 1 ... 13
+          return this.isValid(card,0,51) &&
+          Math.floor(card/4) + 1;
+	},

```